### ISSUE TITLE
Example only shows garbage

### ISSUE THREAD

**Issue by iprognos** [jvandemo/angular-contentful#6]
I have implemented the all content example:

 <section contentful-entries="">
      <p>{{ $contentfulEntries.total }} items found:</p>
      <ul>
        <li ng-repeat="entry in $contentfulEntries.items">
          <pre>{{ entry.sys.id }}</pre>
        </li>
      </ul>
    </section>

There is only a list of garbage. How do I get some actual content from this?

**Comment by jvandemo** [comment]
@iprognos &mdash; Can you post your markup and the result you get?

For example:

``` xml
<ul contentful-entries>
  <li ng-repeat="entry in $contentfulEntries.items">
    {{ entry.fields.name }}
  </li>
</ul>
```

loops over all entries in your Contentful space, displaying the `name` field.

Can you post your JavaScript and HTML? Thanks!

**Comment by iprognos** [comment]
Now I have managed to show the Hemingway blog post:

<h2>Get all entries</h2>

```
<section contentful-entries="" >
  <p>{{ $contentfulEntries.total }} items found:</p>
  <ul>
    <li ng-repeat="entry in $contentfulEntries.items">
      <pre>{{ entry.fields.createdEntries[0].fields.body }}</pre>
    </li>
  </ul>
```

It shows the blog post and then 5 empty ones.

This feel a bit wrong. But I am learning.

**Comment by iprognos** [comment]
This works:

 <li ng-repeat="entry in $contentfulEntries.items">
    {{ entry.fields.name }}
  </li>

Result:

Lewis Carroll
Mike Springer

**Comment by iprognos** [comment]
I have to learn markdown to make code viewable.... Strange that one is working but not the last one.

**Comment by iprognos** [comment]
I am trying to understand what the id are so I can have a clickable list of Blog entries...

**Comment by jvandemo** [comment]
@iprognos &mdash; Angular-contentful provides you with the data that is received from Contentful.

You can read more about the JSON format right here on the Contentful website: [https://www.contentful.com/developers/documentation/content-delivery-api/#json](https://www.contentful.com/developers/documentation/content-delivery-api/#json).

There you can read on system properties (`entry.sys.*`) and fields (`entry.fields.*`) of an entry.

That is how Contentful structures their documents and is not specific to angular-contentful.

Hope that makes sense? Thanks!

**Comment by iprognos** [comment]
Yes, great!

I am now strugling to get a drill down view of blog posts.

I pass the somewhat no seo friendly Id to my details view:

Then I run your directive:  <pre contentful-entry="fdfjhr89734598f">{{ $contentfulEntry }}</pre>

And instead of a hardcoded ID I pass it dynamicly:

 <pre contentful-entry="{{id}}">{{ $contentfulEntry }}</pre>

But I have not yet figured out the right syntax, I have tried with contentful-entry="{{id}}" and contentful-entry="({id})" and contentful-entry="id" and contentful-entry="{id}".

The request to contentful is soething like this, you see whats wrong in the URL:
 https://cdn.contentful.com/spaces/1jdsnorf6e9t/entries/%7B%7Bid%7D%7D?access_token=02d0e52b0cfb62cb9fc011cfe849191fd58a9abb285a388afd68720cf9ab1df7

**Comment by iprognos** [comment]
Not able to pass it dynamicly. I have managed to pass the id succesfully to my details view and I have it in the scoope: $scope.id and I can view it in the details page with this: {{id}} .

But I can not pass in the {{id}} in the contentful-entry handler. I have tried a lot of syntax on the id. 

I am used to pass in id this way: ui-sref=".detail({id: show.name})"

**Comment by jvandemo** [comment]
@iprognos &mdash; `contentful-entry` only accepts a string indeed. I can make it dynamic but that would introduce a breaking change.

If I were you I would resolve the entry in the route's `resolve` using the `contentful` service:

```
// add resolve in your UI-router route:
resolve: {
  entry: getEntry
}

function getEntry($stateParams, contentful){
  return contentful
    .entry($stateParams.id)
    .then(function(response){
      return response.data;
    });
}

getEntry.$inject = ['$stateParams', 'contentful'];
```

Then you get the `entry` injected in your controller and you can:

``` javasript
function YourController($scope, entry){
  $scope.entry = entry;
}
```

to make the entry accessible in your view.

> I'm typing this out of my head. If you can create a plunker I am happy to help you out there.

Hope that helps!

**Comment by iprognos** [comment]
Yes that could be a way to do it.

I did it this way:

I have this in my ui-router:

.state('app.blog.detail', {
        url: '/:id',
        views: {
            'content@': {
                templateUrl: 'app/templates/partials/blog-detail.html',
                controller: 'blogDetailsController'  
            }
        }
    });

I have a blogDetailsController:

app.controller('blogDetailsController', ['$scope','$stateParams', 'contentful', function($scope, $stateParams, contentful ) {
        contentful
          .entry($stateParams.id)
          .then(
          function (response) {
            $scope.$contentfulEntry = response.data;
            console.log($scope.$contentfulEntry);  
          },
          function(){
            $scope.$contentfulEntry = {};
              });
}]);

And in my blog-detail.html I have:

<div class="container">
   <h2>Get one entry</h2>
    <p>{{ $contentfulEntry  }}</p><hr>
   ID: {{id}}
    
</div>

The problem now is that I can't get the body from the blog post:

{{ $contentfulEntry.fields.createdEntries[0].fields.body }}

Its not there.

I have this in the $contentfulEntry:

{"name":"Lewis Carroll","website":"http://en.wikipedia.org/wiki/Lewis_Carroll","profilePhoto":{"sys":{"id":"2ReMHJhXoAcy4AyamgsgwQ","type":"Link","linkType":"Asset"}},"biography":"Charles Lutwidge Dodgson (27 January 1832 – 14 January 1898), better known by his pen name, Lewis Carroll, was an English writer, mathematician, logician, Anglican deacon and photographer. \n\nHis most famous writings are Alice's Adventures in Wonderland, its sequel Through the Looking-Glass, which includes the poem Jabberwocky, and the poem The Hunting of the Snark, all examples of the genre of literary nonsense. \n\nHe is noted for his facility at word play, logic, and fantasy. There are societies in many parts of the world (including the United Kingdom, Japan, the United States, and New Zealand[3]) dedicated to the enjoyment and promotion of his works and the investigation of his life. In 1982, his great-nephew unveiled his memorial stone in Poets' Corner, Westminster Abbey.","createdEntries":[{"sys":{"id":"1asN98Ph3mUiCYIYiiqwko","type":"Link","linkType":"Entry"}}]}

**Comment by jvandemo** [comment]
@iprognos &mdash; Ah, if your entry has links to other entries, you should resolve them.

However, if you use `contentful.entries(querystring)` the resolution will be done automatically for you.

So can you try to use something like `contentful.entries('sys.id=id-of-your-entry')` instead of `contentful.entry(id)`?

> Again I'm typing this without access to the docs so make sure to check the actual syntax. Thx!

**Comment by iprognos** [comment]
Ah ok, I will try that. I did not know the syntax for the query string so I avoided it.

**Comment by iprognos** [comment]
Puh.. Got it working:

blogdetails.controler.js:

app.controller('blogDetailsController', ['$scope','$stateParams', 'contentful', function($scope, $stateParams, contentful ) {

```
    contentful
    .entries('sys.id=' + $stateParams.id )
    .then(
      function (response) {
        $scope.$contentfulEntries = response.data;
          console.log('Response');
          console.log($scope.$contentfulEntries);
      },
      function(){
        $scope.$contentfulEntries = {
          limit: 0,
          skip: 0,
          total: 0,
          items: []
        };
      }
    );

}]);
```

blogdetail.html:

<div class="container">
 
<h1>Directives demo</h1>
    <p>This simple demo uses the following Contentful configuration:</p>
   
   <h2>Get one entry</h2>
    {{ $contentfulEntries.items[0].fields.createdEntries[0].fields.body}}
  
    
</div>

**Comment by iprognos** [comment]
Is it possible to get an entry by slug instead of ID, it´s not very SEO friendly to have a ID in the URL?

I searched the API documentation and could not find any comprehensible. Then I tried with: .entries('fields.slug=down-the-rabbit-hole')

but of course it did not work.

**Comment by jvandemo** [comment]
@iprognos &mdash; Yes, you can specify a query string like:

`content_type=cat&fields.likes=lasagna`

You can see the exact documentation [right here](https://www.contentful.com/developers/documentation/content-delivery-api/#search).

Let me know if it works, ok?

**Comment by iprognos** [comment]
No does not work. query=all works best, then I get three entries and avoid listing categories and authors.

**Comment by jvandemo** [comment]
Closed, continued in #7.
